python_modules/libraries/dagster-embedded-elt/dagster_embedded_elt/sling/asset_decorator.py

```diff
@@ -0,0 +1,181 @@
+        Dagster AssetKey for a stream as follows:
+
+        public.users:
+           meta:
+             dagster:
+               asset_key: "mydb_users"
+
+        Args:
+            stream_name (str): The name of the stream.
+
+        Returns:
+            AssetKey: The Dagster AssetKey for the replication stream.
+
+        Examples:
+            Using a custom mapping for streams:
+
+            class CustomSlingTranslator(DagsterSlingTranslator):
+                @classmethod
+                def get_asset_key_for_target(cls, stream_name: str) -> AssetKey:
+                    map = {"stream1": "asset1", "stream2": "asset2"}
+                    return AssetKey(map[stream_name])
+        """
+        components = cls.sanitize_stream_name(stream_name).split(".")
+        return AssetKey([target_prefix] + components)
+
+    @public
+    @classmethod
+    def get_deps_asset_key(cls, stream_name: str) -> AssetKey:
+        """A function that takes a stream name from a Sling replication config and returns a
+        Dagster AssetKey for the dependencies of the replication stream.
+
+        By default, this returns the stream name. For example, a stream
+        named "public.accounts" will create an AssetKey named "target_public_accounts" and a
+        depenency AssetKey named "public_accounts".
+
+        Override this function to customize how to map a Sling stream to a Dagster AssetKey.
+        Alternatively, you can provide metadata in your Sling replication config to specify the
+        Dagster AssetKey for a stream as follows:
+
+        public.users:
+           meta:
+             dagster:
+               deps: "sourcedb_users"
+
+        Args:
+            stream_name (str): The name of the stream.
+
+        Returns:
+            AssetKey: The Dagster AssetKey dependency for the replication stream.
+
+        Examples:
+            Using a custom mapping for streams:
+
+            class CustomSlingTranslator(DagsterSlingTranslator):
+                @classmethod
+                def get_deps_asset_key(cls, stream_name: str) -> AssetKey:
+                    map = {"stream1": "asset1", "stream2": "asset2"}
+                    return AssetKey(map[stream_name])
+
+
+        """
+        components = cls.sanitize_stream_name(stream_name).split(".")
+        return AssetKey(components)
+
+
+def sling_assets(
+    *,
+    replication: SlingReplicationParam,
+    dagster_sling_translator: DagsterSlingTranslator = DagsterSlingTranslator(),
+    deps: Optional[Iterable[CoercibleToAssetDep]] = None,
+    name: Optional[str] = None,
+    partitions_def: Optional[PartitionsDefinition] = None,
+    backfill_policy: Optional[BackfillPolicy] = None,
+    op_tags: Optional[Mapping[str, Any]] = None,
+    group_name: Optional[str] = None,
+    code_version: Optional[str] = None,
+    freshness_policy: Optional[FreshnessPolicy] = None,
+    auto_materialize_policy: Optional[AutoMaterializePolicy] = None,
```

Comment:
These should just live on the translator for consistency, even though we don't offer subsetting support.

I assume code version could be inferred on behalf of the user as the hash of the config.